src/mavlink_stream.c

```diff
@@ -96,16 +102,16 @@ void mavlink_stream(uint32_t time_us)
   {
     if (mavlink_streams[i].period_us && time_us - mavlink_streams[i].last_time_us >= mavlink_streams[i].period_us)
     {
-      mavlink_streams[i].last_time_us = time_us;
+      // if we took too long, set the last_time_us to be where it should have been
+      mavlink_streams[i].last_time_us = time_us - (time_us - mavlink_streams[i].last_time_us-mavlink_streams[i].period_us);
       mavlink_streams[i].send_function();
     }
   }
 }
 
 void mavlink_stream_set_rate(mavlink_stream_id_t stream_id, uint32_t rate)
 {
-
-  mavlink_streams[stream_id].period_us = (rate == 0 ? 0 : 1e6 / rate);
+  mavlink_streams[stream_id].period_us = (rate == 0 ? 0 : (int32_t)((1000000.0f) / (float)rate));
```

Comment:
So... maybe I'm just being stupid and missing something, but I don't think this helps us at all. Converting to floats before doing the division will mean the remainder gets put into the stuff after the decimal point, but then casting back to an int will get rid of all that anyway. Isn't it equivalent to integer division?